swap/swap.go

```diff
@@ -577,27 +595,37 @@ func (s *Swap) getContractOwner(ctx context.Context, address common.Address) (co
 	return contr.Issuer(nil)
 }
 
-func promptInitialDepositAmount() (uint64, error) {
-	// need to prompt user for initial deposit amount
-	// if 0, can not cash in cheques
+func (s *Swap) promptDepositAmount() (*big.Int, error) {
+	// retrieve available balance
+	availableBalance, err := s.AvailableBalance()
+	if err != nil {
+		return nil, err
+	}
+	balance, err := s.contract.BalanceAtTokenContract(nil, s.owner.address)
+	if err != nil {
+		return nil, err
+	}
+	// log available balance and ERC20 balance
+	swapLog.Info("Balance information", "chequebook available balance", availableBalance, "ERC20 balance", balance)
+	promptMessage := fmt.Sprintf("Please provide the amount in HONEY which will deposited to your chequebook: ")
+	// need to prompt user for deposit amount
 	prompter := console.Stdin
-
 	// ask user for input
-	input, err := prompter.PromptInput("Please provide the amount in Wei which will deposited to your chequebook upon deployment: ")
+	input, err := prompter.PromptInput(promptMessage)
 	if err != nil {
-		return 0, err
+		return big.NewInt(0), err
```

Comment:
again, don't create instances for zero values

Reply:
Ok!